jsk_arc2017_baxter/euslisp/lib/baxter-interface.l

```diff
@@ -0,0 +1,208 @@
+    (&optional (arm :arms))
+    (dolist (l/r (if (eq arm :arms) (list "left" "right") (list (arm2str arm))))
+      (ros::publish
+        (format nil "/gripper_front/limb/~a/servo/torque" l/r)
+        (instance std_msgs::Bool :init :data nil))))
+  (:graspingp
+    (arm)
+    (let (topic)
+      (if (ros::get-param "/apc_on_gazebo" nil)
+        (progn
+          (setq topic (format nil "/robot/~a_vacuum_gripper/grasping" (arm-to-str arm)))
+          (send (one-shot-subscribe topic std_msgs::Bool) :data)
+          )
+        (progn
+          (setq topic (format nil "gripper_front/limb/~a/pressure/state" (arm-to-str arm)))
+          (< (send (one-shot-subscribe topic std_msgs::Float64) :data)
+             (cond
+               ((eq arm :rarm) rarm-pressure-threshold-)
+               ((eq arm :larm) larm-pressure-threshold-)))))))
+  (:arm-potentio-vector
+    (arm)
+    (case arm
+      (:larm (subseq (send *ri* :state :potentio-vector) 1 9))
+      (:rarm (subseq (send *ri* :state :potentio-vector) 9 17))
+      (t nil)))
+  (:tf-pose->coords
+    (frame_id pose)
+    (let ((base-name (send (send (send self :robot) :base_lk) :name)))
+      (send (send tfl- :lookup-transform base-name frame_id (ros::time 0))
+            :transform (ros::tf-pose->coords pose))))
+  (:fold-pose-back
+    (&optional (arm :arms))
+    (dolist (tmp-arm (if (eq arm :arms) (list :rarm :larm) (list arm)))
+      (send (send self :robot) :fold-pose-back tmp-arm))
+    (send self :send-av))
+  (:send-av
+    (&optional (tm 3000) (ctype nil))
+    (send self :angle-vector (send *baxter* :angle-vector) tm ctype))
+  (:wait-interpolation-until-grasp
+    (arm)
+    (when (send self :simulation-modep)
+      (return-from :wait-interpolation-until-grasp (send self :wait-interpolation))
+      )
+    (while (send self :interpolatingp)
+           (if (send self :graspingp arm)
+             (progn
+               (ros::ros-info "[:wait-interpolation-until-grasp] Grasping detected. Cancel angle vector: ~a" arm)
+               (send self :cancel-angle-vector)))))
+  (:calib-pressure-threshold
+    (&optional (arm :arms))
+    (send self :start-grasp arm)
+    (dolist (l/r (if (eq arm :arms) (list :rarm :larm) (list arm)))
+      (let ((min-pressure)
+            (topic (format nil "/gripper_front/limb/~a/pressure/state" (arm-to-str l/r))))
+        (dotimes (i 7)
+          (let ((pressure (send (one-shot-subscribe topic std_msgs::Float64) :data)))
+            (when (or (null min-pressure) (< pressure min-pressure)) (setq min-pressure pressure)))
+          (unix::sleep 1))
+        (cond ((eq l/r :larm)
+               (setq larm-pressure-threshold- (- min-pressure 5)))
+              ((eq l/r :rarm)
+               (setq rarm-pressure-threshold- (- min-pressure 10))))))
+    (send self :stop-grasp arm)
+    (ros::ros-info "[:calib-pressure-threshold] Threshold r: ~a l: ~a"
+                   rarm-pressure-threshold- larm-pressure-threshold-))
+  (:spin-off-by-wrist
```

Comment:
`:spin-off-by-wrist` is just a motion, so I thought it is better to put it in `baxter-interface`